web/src/pages/apd/new/ApdNew.js

```diff
@@ -0,0 +1,562 @@
+    if (new Date().getMonth() > 8) {
+      return year + 1;
+    }
+    return year;
+  })();
+
+  const businessAreaOptions = {
+    waiverSupport: false,
+    assetVerification: false,
+    claimsProcessing: false,
+    decisionSupport: false,
+    electronicVisitVerify: false,
+    encounterProcessingSystem: false,
+    financialMangement: false,
+    healthInfoExchange: false,
+    longTermServiceSupport: false,
+    memberManagement: false,
+    pharmacyBenefitManagement: false,
+    programIntegrity: false,
+    providerManagement: false,
+    thirdPartyLiability: false,
+    other: false
+  };
+
+  const updateTypes = {
+    annualUpdate: false,
+    asNeededUpdate: false
+  };
+
+  const yearOptions = [thisFFY, thisFFY + 1, thisFFY + 2].map(y => `${y}`);
+  const [apdType, setApdType] = useState('');
+  const [businessAreas, setBusinessAreas] = useState(businessAreaOptions);
+  const [isLoading, setIsLoading] = useState(false);
+  const [typeStatus, setTypeStatus] = useState(updateTypes);
+  const [years, setYears] = useState(yearOptions.slice(0, 2));
+
+  const apdTypeChoices = [
+    {
+      label: 'HITECH IAPD',
+      labelClassName: 'label-extended',
+      value: 'HITECH',
+      hint: 'Health Information Techology for Economic and Clinical Health Implementation APD',
+      checked: value
+    },
+    {
+      label: 'MMIS IAPD',
+      labelClassName: 'label-extended',
+      value: 'MMIS',
+      hint: 'Medicaid Management Information System Implementation APD',
+      checked: value
+    }
+  ];
+  const [apdChoices, setApdChoices] = useState(apdTypeChoices);
+
+  useEffect(() => {
+    if (!enableMmis) {
+      apdChoices.pop();
+      apdChoices[0].checked = true;
+      setApdChoices(apdChoices);
+      setApdType('HITECH');
+      setValue('apdType', 'HITECH', { shouldValidate: true });
+    }
+  }, [apdChoices]);
+
+  const {
+    control,
+    setValue,
+    getValues,
+    formState: { errors, isDirty, isValid }
+  } = useForm({
+    defaultValues: {
+      years: years
+    },
+    mode: 'all',
+    reValidateMode: 'all',
+    resolver: joiResolver(newApdSchema)
+  });
+
+  if (isLoading) {
+    return (
+      <div id="start-main-content">
+        <Loading>Loading your APD</Loading>
+      </div>
+    );
+  }
+
+  const yearChoices = yearOptions.map((year, index) => ({
+    defaultChecked: years.includes(year),
+    apdname: `years[${index}]`,
+    label: year,
+    value: year
+  }));
+
+  function updateArray(array, name, setAction, value) {
+    let indexPrime = array.indexOf(value),
+      newArray = array;
+    if (indexPrime > -1) {
+      newArray.splice(indexPrime, 1);
+      setValue(name, newArray, { shouldValidate: true });
+      setAction(newArray);
+    } else if (indexPrime === -1) {
+      newArray.push(value);
+      setValue(name, newArray, { shouldValidate: true });
+      setAction(newArray);
+    }
+  }
+
+  const createNew = () => {
+    let values = getValues();
```

Comment:
You need to change up the data and format that you are sending to create because the values won't be saved in mongo unless the data format matches. Right now what you are sending back looks like this
```json
{
  "years": ["2023", "2024"],
  "apdType": "MMIS",
  "name": "Test",
  "mmisUpdate": "no",
  "apdOverview": {
    "medicaidBA": [
      "assetVerification",
      "electronicVisitVerify",
      "longTermServiceSupport"
    ]
  },
  "updateStatus": {
    "updateList": [],
    "typeStatus": { "annualUpdate": false, "asNeededUpdate": false }
  }
}
```
but the schema expects it to look like
```json
{
  "years": ["2023", "2024"],
  "apdType": "MMIS",
  "name": "Test",
  "apdOverview": {
    "isUpdateAPD": false,
    "annualUpdate": false,
    "asNeededUpdate": false
  },
  "medicaidBusinessAreas": {
    "waiverSupportSystems": false,
    "assetVerificationSystem": true,
    "claimsProcessing": false,
    "decisionSupportSystemDW": false,
    "electronicVisitVerification": true,
    "encounterProcessingSystemMCS": false,
    "financialManagement": false,
    "healthInformationExchange": false,
    "longTermServicesSupports": true,
    "memberManagement": false,
    "pharmacyBenefitManagementPOS": false,
    "programIntegrity": false,
    "providerManagement": false,
    "thirdPartyLiability": false,
    "other": false,
    "otherMedicaidBusinessAreas": ""
  }
}
```
you should be able to do it by changing it to 
```javascript
const { years, name, mmisUpdate } = getValues();
const apdValues = {
  years,
  name,
  apdType,
  apdOverview: {
    ...typeStatus
  }
};
if (apdType === APD_TYPE.HITECH) {
  apdValues.apdOverview.isUpdateAPD = true;
}
if (apdType === APD_TYPE.MMIS) {
  apdValues.apdOverview.isUpdateAPD = (mmisUpdate === 'yes');
  apdValues.apdOverview.medicaidBusinessAreas = businessAreas;
}
setIsLoading(true);
create(apdValues);
```
you also need to go through all the businessArea field names and make sure they match the name that the backend is expecting.